### PR TITLE
Change w3c correlation context to custom header

### PR DESCRIPTION
**Pull request by dyladan**
After discussions with the W3C Distributed Tracing Working Group, it has been determined that the Correlation-Context header is not ready for implementation and is being actively changed. As recommended by the working group, we should implement our own header until the specification is an official recommendation.

/cc @tedsuo @mwear

### CODE REVIEW

**Comment by dyladan** [comment]
> I am fine with this. QQ: should this be part of the default API?

Think this discussion is in #428

**Comment by bogdandrutu** [comment]
@SergeyKanzhelev please review. also @open-telemetry/specs-approvers  :)

**Comment by dyladan** [comment]
The W3C Distributed Tracing WG is discussing this issue this week. Please do not merge this PR yet.

**Comment by dyladan** [comment]
@tedsuo see https://github.com/w3c/correlation-context/issues/18

The WG decided not to go with IETF dict, and since this isn't merged yet it isn't too late to change

**Comment by dyladan** [comment]
@yurishkuro would appreciate a ✅ here

**Comment by yurishkuro** [comment]
btw, why not `otcorreations` to match the Correlations API?

**Comment by dyladan** [comment]
Originally I had been trying to keep the header name as similar as possible to the name I thought the w3c would likely end up with. Because of the recent naming discussions, I think it's clear that we should just use whatever header name makes sense for us.

**Comment by dyladan** [comment]
We need to decide on something here so we can get this merged.

React to this message to vote

1. `otcorrelationcontext` 😄 
2. `otbaggage` 🎉 
3. `otcorrelations` 👀

**Comment by dyladan** [comment]
Rebased with name `otbaggage` PTAL and merge when you're happy

**Comment by MrAlias** [comment]
@dyladan: it seems like the above poll shows `otcorrelations` as the prefered header name for the time being. Why did we switch to `otbaggage`?

File: specification/correlationcontext/api.md [hunk 1]
```diff
@@ -102,6 +101,37 @@ OPTIONAL parameters:
...
+{
+  "user": "foo@example.com",
+  "name": "Example Name"
+}
+```
+
+Header:
+
+```
+otcorrelcationcontext: user=foo%40example.com,name=Example%20Name
```
Comment (MrAlias) [inline specification/correlationcontext/api.md:132]: We'll want this to match the above specified header name.

**Comment by dyladan** [comment]
@MrAlias oh man I just had a brain fart I guess. updating now.

**Comment by dyladan** [comment]
Can't see why the build is failing. Just getting a spinner in circle ci that says 0 jobs in the workflow.

image

**Comment by MrAlias** [comment]
What I'm seeing:

```
#!/bin/bash -eo pipefail
make markdown-lint
make enforce-markdown-link-check
markdownlint -c .markdownlint.yaml '**/*.md'
markdown-link-check FAILED => errors:

Run 'make markdown-link-check' to see the errors
Makefile:30: recipe for target 'enforce-markdown-link-check' failed
make: *** [enforce-markdown-link-check] Error 1

Exited with code exit status 2
```

**Comment by MrAlias** [comment]
That Makefile command is not really helpful. Running manually:

```
$ find . -name \*.md -exec markdown-link-check {} 2>&1 \;
...
FILE: ./specification/context/api-propagators.md
[✓] #overview
[✓] #http-text-format
[✓] #fields
[✓] #inject
[✓] #setter
[✓] #set
[✓] #extract
[✓] #getter
[✓] #get
[✓] #composite-propagator
[✓] #global-propagators
[✖] api-correlationcontext.md#serialization

12 links checked.

ERROR: 1 dead links found!
[✖] api-correlationcontext.md#serialization → Status: 400
...
```

**Comment by dyladan** [comment]
@lizthegrey would you rather we use `otelbaggage`? It's only another 2 bytes on the header so I don't think we need to worry too much about payload size. I typically think of `ot` as OpenTelemetry not OpenTracing, but that's just me and I know not everyone feels that way.

**Comment by yurishkuro** [comment]
there was never `otbaggage` in opentracing, so I don't think there's much confusion.

**Comment by mwear** [comment]
There seems to be some continued confusion about what we're actually changing the header name to. I see discussions about `otbaggage`, but this PR currently changes it to `otcorrelations`. I think that is due to https://github.com/open-telemetry/opentelemetry-specification/pull/517#issuecomment-611508995.

**Comment by dyladan** [comment]
> There seems to be some continued confusion about what we're actually changing the header name to. I see discussions about `otbaggage`, but this PR currently changes it to `otcorrelations`. I think that is due to [#517 (comment)](https://github.com/open-telemetry/opentelemetry-specification/pull/517#issuecomment-611508995).

Yes my fault. `otcorrelations` is the name we decided on in my informal poll, I just can't get `baggage` out of my head because that's what the WG is going towards.

**Comment by dyladan** [comment]
Is this waiting on something to merge?

**Comment by MrAlias** [comment]
@open-telemetry/technical-committee can we merge this so language SIGs have direction on this?

**Pull request merged**